client/my-sites/domains/components/form/hidden-input.jsx

```diff
@@ -2,44 +2,42 @@
  * External dependencies
  */
 import { isEmpty } from 'lodash';
-const React = require( 'react' );
+import React from 'react';
 
 /**
  * Internal dependencies
  */
-var Input = require( './input' );
+import Input from './input';
 
-module.exports = React.createClass( {
-	displayName: 'HiddenInput',
+export default class extends React.Component {
+	static displayName = 'HiddenInput';
 
-	componentWillReceiveProps: function( nextProps ) {
+	state = {
+		toggled: false
+	};
+
+	componentWillReceiveProps( nextProps ) {
 		if ( ! this.state.toggled && ! isEmpty( nextProps.value ) ) {
 			this.setState( { toggled: true } );
 		}
-	},
-
-	getInitialState: function() {
-		return {
-			toggled: false
-		};
-	},
+	}
 
-	componentDidUpdate: function( prevProps, prevState ) {
+	componentDidUpdate( prevProps, prevState ) {
 		// Focus the input only when the user explicitly clicked the toggle link
 		if ( ! prevState.toggled && this.state.toggled && prevProps.value === this.props.value ) {
 			this.refs.input.focus();
 		}
-	},
+	}
 
-	handleClick: function( event ) {
+	handleClick = event => {
```

Comment:
I think the `handleClick = ( event ) => { }` notation (putting the parameters in parenthesis) is more explicit - it's easier to understand that we're dealing with an arrow function. While `handleClick = event => {}` looks a bit too much like simple assignment to me.